src/pint/models/dispersion_model.py

```diff
@@ -320,42 +320,63 @@ def __init__(self):
                 description="Dispersion measure",
             )
         )
+
+        self.add_DMX_range(None,None,dmx=0,frozen=False,index=1)
+
+        self.dm_value_funcs += [self.dmx_dm]
+        self.set_special_params(["DMX_0001", "DMXR1_0001", "DMXR2_0001"])
+        self.delay_funcs_component += [self.DMX_dispersion_delay]
+
+    def add_DMX_range(self,mjd_start,mjd_end,index,dmx=0,frozen=False):
+   #### Setting up the DMX title convention. If index is None, want to increment the current max DMX index by 1. 
+        if index is None:
+            index=self.get_prefix_mapping_component('DMX_').max()+1
+        i="DMX_"+f"{int(index):04d}"
+
+   #### Check to see if the user-input index is already assigned to a different DMX component. Throw error if it is.
+        if int(index) in self.get_prefix_mapping_component('DMX_'): 
+            raise ValueError("Index '%s' is already in use in this model. Please choose another." %index)
+
+   #### Add three DMX parameters (start MJD, end MJD, DMX value)
         self.add_param(
             prefixParameter(
-                name="DMX_0001",
+                name="DMX_"+i,
```

Comment:
This will not work since line 334 already have 'DMX_' in `i`. `i` should just be `000N`